functions/go/set-namespace/transformer/namespace.go

```diff
@@ -18,206 +18,311 @@ import (
 	"strings"
 
 	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func SetNamespace(rl *fn.ResourceList) (bool, error) {
-	tc := NamespaceTransformer{}
+// Run provides the main workflow to update the ResourceList.Items "namespace" value.
+func Run(rl *fn.ResourceList) (bool, error) {
```

Comment:
Why changing the name?

Reply:
Thanks for asking. 😄  I want to use the `SetNamespace` as the transformer name which is also the function config kind. So the transformer is served as the typed functionConfig parser at the same time. The advantage is that the `As` method can be called in a single line (well I believe you must already use this or something similar) 
```go
func (t *SetNamespace) Config(fnConfig *fn.KubeObject) {
     fnConfig.As(t)
}